twine/commands/upload.py

```diff
@@ -257,10 +258,14 @@ def upload(dists, repository, sign, identity, username, password, comment,
         if resp.is_redirect:
             raise exc.RedirectDetected(
                 ('"{0}" attempted to redirect to "{1}" during upload.'
-                 ' Aborting...').format(config["respository"],
+                 ' Aborting...').format(config["repository"],
                                         resp.headers["location"]))
         # Otherwise, raise an HTTPError based on the status code.
-        resp.raise_for_status()
+        if resp.status_code == 400 and skip_existing:
```

Comment:
I don't disagree. It looks like the reason part of the status line returns something like:

```
(Pdb) p resp.reason
'A file named "requests_toolbelt-0.4.0-py2.py3-none-any.whl" already exists for  requests-toolbelt-0.4.0. To fix problems with that file you should create a new release.'
```

Again this seems kind of fragile, but may as well test for it. (To be clear, the behaviour of PyPI in this matter can change very drastically in the coming months given the timeline for Warehouse's release.)